lib/crowbar/client/request/network.rb

```diff
@@ -0,0 +1,155 @@
+#
+# Copyright 2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  module Client
+    class Request
+      module Network
+        extend ActiveSupport::Concern
+
+        included do
+          def network_allocate_virtual(
```

Comment:
Should be named  `network_allocate_virtual_ip`; there could be other virtual things.